cupy/manipulation/rearrange.py

```diff
@@ -1,6 +1,76 @@
+import itertools
+import numpy
+
+
 import cupy
 from cupy import core
 
+try:
+    from numpy.core.numeric import normalize_axis_tuple
```

Comment:
We don't think we should use `numpy.core` module, because it's an undocumented module.
We should avoid using copy-pasted NumPy code either, because the feature is simple enough to be implemented by ourselves.

Could you either re-implement it, or use (or extend) the existing normalization logic?
The existing implementation is below:
https://github.com/cupy/cupy/blob/0fe2fd9a4ab1d7a1f14aceba780ca59828dd8c75/cupy/core/reduction.pxi#L91)

Reply:
I found that in numpy <=1.12 an out-of-bounds axis returns `ValueError`, but starting from 1.13, it returns a newly defined `numpy.AxisError` instead (where `numpy.AxisError` inherits from both `ValueError` and `IndexError`).  That is why the test cases with `accept_error='numpy.AxisError'` specify numpy>=1.13. 

Aside from changing the issue with the error type, the existing `_get_axis` logic should work here.

Reply:
Ah, we have `core.core._AxisError` for that reason I think. How about raising it?
That way I think `accept_error` can be omitted.

Reply:
Okay, done.  I guess I should have grepped for `AxisError` first.  I just saw that `cupy.AxisError` did not exist (perhaps it should, but that is separate from the issue in this PR)